node/src/blockchain/blockchain_interface/blockchain_interface_web3/utils.rs

```diff
@@ -302,41 +328,47 @@ pub fn send_payables_within_batch(
                 Ok(merged_output_data(
                     batch_response,
                     hashes_and_paid_amounts_ok,
-                    accounts,
+                    accounts.into(),
                 ))
             }),
     )
 }
 
 pub fn create_blockchain_agent_web3(
+    qualified_payables: QualifiedPayablesRawPack,
     gas_limit_const_part: u128,
     blockchain_agent_future_result: BlockchainAgentFutureResult,
     wallet: Wallet,
     chain: Chain,
-) -> Box<dyn BlockchainAgent> {
-    Box::new(BlockchainAgentWeb3::new(
+) -> (Box<dyn BlockchainAgent>, QualifiedPayablesRipePack) {
+    let transaction_fee_balance_in_minor_units =
+        blockchain_agent_future_result.transaction_fee_balance;
+    let masq_token_balance_in_minor_units = blockchain_agent_future_result.masq_token_balance;
+    let cons_wallet_balances = ConsumingWalletBalances::new(
+        transaction_fee_balance_in_minor_units,
+        masq_token_balance_in_minor_units,
+    );
+    BlockchainAgentWeb3::new(
```

Comment:
We discussed that you should use another method to generate the `QualifiedPayablesRipePack`. Another improvement that we can make is use another mediator structure.

```rust
pub struct BlockchainAgentWeb3WithPayables {
    agent: BlockchainAgentWeb3,
    payables: QualifiedPayablesRipePack,
}
```